src/sortStudents.ts

```diff
@@ -1,16 +1,61 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, prev) => sum + prev, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student [],
+  sortBy: SortType,
+  order: SortOrder,
+): Student [] {
+  const studentsClone = [...students];
+
+  return studentsClone.sort((studentA, studentB) => {
+    switch (sortBy) {
+      case SortType.Name:
+
+        return order === 'asc'
+          ? studentA.name.localeCompare(studentB.name)
+          : studentB.name.localeCompare(studentA.name);
+
+      case SortType.Surname:
+
+        return order === 'asc'
+          ? studentA.surname.localeCompare(studentB.surname)
+          : studentB.surname.localeCompare(studentA.surname);
+
+      case SortType.Age:
+        return order === 'asc'
+          ? studentA.age - studentB.age
+          : studentB.age - studentA.age;
```

Comment:
_comment_
I think it is better to use your Enum variable as keys in this case as you created it to be free of hardcoding.
```suggestion
          ? studentA[SortType.Age] - studentB[SortType.Age]
          : studentB[SortType.Age] - studentA[SortType.Age];
```

Reply:
I'm not sure this is a good point